examples/webgpu_materials.html

```diff
@@ -0,0 +1,202 @@
+		<link type="text/css" rel="stylesheet" href="main.css">
+	</head>
+	<body>
+		<div id="info">
+			<a href="https://threejs.org" target="_blank" rel="noopener">three.js</a> - webgpu materials
+		</div>
+
+		<script type="module">
+
+			import * as THREE from '../build/three.module.js';
+
+			import WebGPURenderer from './jsm/renderers/webgpu/WebGPURenderer.js';
+			import WebGPU from './jsm/renderers/webgpu/WebGPU.js';
+
+			import { TeapotGeometry } from './jsm/geometries/TeapotGeometry.js';
+
+			import AttributeNode from './jsm/renderers/nodes/core/AttributeNode.js';
+			import PositionNode from './jsm/renderers/nodes/accessors/PositionNode.js';
+			import NormalNode from './jsm/renderers/nodes/accessors/NormalNode.js';
+			import TextureNode from './jsm/renderers/nodes/inputs/TextureNode.js';
+
+			import Stats from './jsm/libs/stats.module.js';
+
+			let stats;
+
+			let camera, scene, renderer;
+
+			const objects = [], materials = [];
+
+			init().then( animate ).catch( error );
+
+			async function init() {
+
+				if ( WebGPU.isAvailable() === false ) {
+
+					document.body.appendChild( WebGPU.getErrorMessage() );
+
+					throw 'No WebGPU support';
+
+				}
+
+				const container = document.createElement( 'div' );
+				document.body.appendChild( container );
+
+				camera = new THREE.PerspectiveCamera( 45, window.innerWidth / window.innerHeight, 1, 2000 );
+				camera.position.set( 0, 200, 800 );
+
+				scene = new THREE.Scene();
+
+				// Grid
+
+				const helper = new THREE.GridHelper( 1000, 40, 0x303030, 0x303030 );
+				helper.material.colorNode = new AttributeNode( 'color', 'vec3' );
+				helper.position.y = - 75;
+				scene.add( helper );
+
+				// Materials
+
+				const textureLoader = new THREE.TextureLoader();
+
+				const texture = textureLoader.load( './textures/uv_grid_opengl.jpg' );
+				texture.wrapS = THREE.RepeatWrapping;
+				texture.wrapT = THREE.RepeatWrapping;
+
+				let material;
+
+				// PositionNode.LOCAL
+				material = new THREE.MeshBasicMaterial();
+				material.colorNode = new PositionNode( PositionNode.LOCAL );
+				materials.push( material );
+
+				// NormalNode.LOCAL
+				material = new THREE.MeshBasicMaterial();
+				material.colorNode = new NormalNode( NormalNode.LOCAL );
+				materials.push( material );
+
+				// NormalNode.WORLD
+				material = new THREE.MeshBasicMaterial();
+				material.colorNode = new NormalNode( NormalNode.WORLD );
+				materials.push( material );
+
+				// NormalNode.VIEW
+				material = new THREE.MeshBasicMaterial();
+				material.colorNode = new NormalNode( NormalNode.VIEW );
+				materials.push( material );
+
+				// TextureNode
+				material = new THREE.MeshBasicMaterial();
+				material.colorNode = new TextureNode( texture );
+				materials.push( material );
+
+				// Opacity
+				material = new THREE.MeshBasicMaterial();
+				material.opacityNode = new TextureNode( texture );
+				material.transparent = true;
```

Comment:
BTW: It's good that you implement transparency settings like so 👍 . This should solve #18038 since the render list is now correct in the first frame.

Reply:
This was not a hated feature in general, it facilitates in most cases ( https://github.com/mrdoob/three.js/pull/16996#issuecomment-514725009 ) . More few months we must abandon the old `NodeMaterial` system and adopt the new one that uses `.transparent`... Should we wait this change only for the new `NodeMaterial` system?

Reply:
Replacing the old material system with the new one makes sense. But maybe it's better to invest a bit more time in developing the new one before doing this step.